includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -1817,6 +1817,10 @@ protected function extract_response_body( $response ) {
 					$response_code
 				);
 			} elseif ( isset( $response_body['error'] ) ) {
+				if ( isset( $response_body['error']['decline_code'] ) && 'fraudulent' === $response_body['error']['decline_code'] ) {
+					WC()->cart->empty_cart();
+					WC()->session->set( 'reload_checkout', true );
```

Comment:
Sorry for being late for the review. What does `WC()->session->set( 'reload_checkout', true );` do?

Reply:
```WC()->session->set( 'reload_checkout', true );```

The front end upon receiving a response will refresh the page. It basically displays the error messages.

Reply:
> Also, we should consider keeping the cart content, would it be possible?

If we trust the fraudulent response then emptying the cart is the best option to do as it forces to perform several steps. 

WP nonce is tied to session and time. We can't regenerate a new nonce that could be used with the existing cart.

Reply:
I see. This could be an option, but we need to do a little bit more here: particularly we need a setting that will enable/disable the protection.

Hence, let's revert the PR and re-open the issue for the next iteeration.